omnibus/files/private-chef-cookbooks/private-chef/libraries/preflight_checks.rb

```diff
@@ -0,0 +1,36 @@
+class PreflightChecks
+  attr_reader :node
+  def initialize(node)
+    @node = node
+  end
+
+  # Any validator is expected to throw a PreflightChecks::ValidationFailed exception if the run
+  # must be stopped.
+  class ValidationFailed < Exception
+
+  end
+
+  # Run our validators to ensure we're in a good state to perform a reconfigure/chef client run.
+  # Stop the run immediately if a validation failure occurs.
+  # We will do so in a way that prevents the default at_exit handlers from being invoked -
+  # which means that the error text is immediately visible to the user, without forcing
+  # them to hunt for it in backscroll.
+  #
+  # Validations are expected to be run after chef-server.rb entries are ingested, but before any
+  # defaults are configured via libraries/private_chef.rb
+  def run!
+    begin
+      PostgresqlPreflightValidator.new(node).run!
```

Comment:
Yeah, there are a few examples of factory-style constructors that collect the relevant classes in Chef.  I'm thinking though it might be better to just keep it super explicit in this case.  A nice list of class instantiations that define the set of checks we are gonna do makes it really easy to see what checks are going to run and in what order.